src/main/java/com/google/devtools/build/lib/rules/genrule/GenRuleBaseRule.java

```diff
@@ -166,8 +166,75 @@ are always generated into a predictable location (available via <code>$(@D)</cod
             </a> in this document for a list of supported values.
           </li>
         </ul>
+        <p>
+        This is the fallback of `cmd_bash`, `cmd_ps` and `cmd_bat`, if none of them are applicable.
+        </p>
+        <!-- #END_BLAZE_RULE.ATTRIBUTE --> */
+        .add(attr("cmd", STRING))
+
+        /* <!-- #BLAZE_RULE(genrule).ATTRIBUTE(cmd_bash) -->
+        The Bash command to run.
+        <p> This attribute has higher priority than `cmd`. The command is expanded and runs in
+            the exact same way as the `cmd` attribute.
+        </p>
+        <!-- #END_BLAZE_RULE.ATTRIBUTE --> */
+        .add(attr("cmd_bash", STRING))
+
+        /* <!-- #BLAZE_RULE(genrule).ATTRIBUTE(cmd_bat) -->
+        The Batch command to run on Windows.
+        <p> This attribute has higher priority than `cmd` and `cmd_bash`. The command runs in
+            the similar way as the `cmd` attribute, with the following differences:
+        </p>
+        <ul>
+          <li>
+            This command only applies on Windows.
+          </li>
+          <li>
+            The command runs with `cmd.exe /c`.
```

Comment:
If you use more flags (as I suggested), remember to update here.

Reply:
Yes! Thanks!